code/controllers/subsystems/news.dm

```diff
@@ -17,60 +17,59 @@
+		UNTIL(initial.is_complete())
+
+		var/datum/http_response/initial_response = initial.into_response()
+		if (initial_response.errored)
+			testing("Errored: [initial_response.error]")
+			log_debug("SSnews: errored: [initial_response.error]")
+			continue
+
+		var/list/forum_topic = initial_response.body
+		var/datum/feed_channel/channel = CreateFeedChannel(
+			forum_topic["title"], "System", TRUE, TRUE, FALSE
+		)
+
+		var/datum/http_request/forum_api/posts = new("forums/topics")
+		posts.prepare_get("[topic_id]/posts", list("sortDir" = "desc", "hidden" = 0, "page" = 1, "perPage" = 10))
+		posts.begin_async()
+		UNTIL(posts.is_complete())
+
+		var/datum/http_response/posts_response = posts.into_response()
+		if (posts_response.errored)
+			log_debug("SSnews: errored getting posts from [topic_id]: [posts_response.error]")
 			continue
 
-		//Load the expired news into the modular computers first
+		var/list/forum_posts = posts_response.body
+
 		var/news_count = 1
-		var/DBQuery/news_query_expired = dbcon.NewQuery("SELECT body FROM ss13_news_stories WHERE deleted_at IS NULL AND channel_id = :channel_id: AND publish_at < NOW() AND (publish_until <= NOW()) AND approved_at IS NOT NULL ORDER BY publish_at DESC")
-		news_query_expired.Execute(list("channel_id" = channel_query.item[1]))
-		while(news_query_expired.NextRow())
-			CHECK_TICK
-			try
-				var/datum/computer_file/data/news_article/news = new()
-				news.filename = "[channel.channel_name] vol. [news_count]"
-				news.archived = 1
-				news.stored_data = news_query_expired.item[1]
-				ntnet_global.available_news.Add(news)
-				news_count += 1
-			catch(var/exception/em)
-				log_debug("SSnews: Error when loading news: [em]")
-
-		var/DBQuery/news_query = dbcon.NewQuery("SELECT body, author, is_admin_message, message_type, ic_timestamp, url FROM ss13_news_stories WHERE deleted_at IS NULL AND channel_id = :channel_id: AND publish_at < NOW() AND (publish_until > NOW() OR publish_until IS NULL) AND approved_at IS NOT NULL ORDER BY publish_at DESC")
-		news_query.Execute(list("channel_id" = channel_query.item[1]))
-		while(news_query.NextRow())
-			CHECK_TICK
-			try
-				SubmitArticle(news_query.item[1], news_query.item[2], channel, null, text2num(news_query.item[3]), news_query.item[4], news_query.item[5])
-				var/datum/computer_file/data/news_article/news = new()
-				news.filename = "[channel.channel_name] vol. [news_count]"
-				news.archived = 0
-				news.stored_data = news_query.item[1]
-				ntnet_global.available_news.Add(news)
-				news_count += 1
-			catch(var/exception/en)
-				log_debug("SSnews: Error when loading news: [en]")
+		for (var/list/post in forum_posts["results"])
+			SubmitArticle(
+				post["content"], GetForumAuthor(topic_id, post["id"]), channel,
+				null, FALSE, "Story", GetForumTimestamp(post["date"])
+			)
+
+			var/datum/computer_file/data/news_article/news = new()
+			news.filename = "[channel.channel_name] vol. [news_count]"
+			news.archived = 0
```

Comment:
It might be interesting to add a few more "archived" posts to the modular computer news browser.